### ISSUE TITLE
SSR issues when using cookie storage

### ISSUE THREAD

**Issue by DavidPhilip** [amplitude/Amplitude-TypeScript#444]
## Expected Behavior
When building the app with nextjs and SSR enabled the app builds without errors.

## Current Behavior
The build for server-side generated pages that rely on Amplitude on the client-side fails with:

```js
TypeError: Cannot read properties of undefined (reading 'cookie')
    at CookieStorage.<anonymous> (/Users/USER_NAME/Documents/PROJECT_NAME/node_modules/@amplitude/analytics-client-common/lib/cjs/storage/cookie.js:81:111)
```

The line in question is this one:
https://github.com/amplitude/Amplitude-TypeScript/blob/46b297c11b5d2a83678824b119191fb955df6096/packages/analytics-client-common/src/storage/cookie.ts#L53

During SSR `globalScope` is defined but it will not have a `document` or `window` defined.

## Possible Solution
Check if  `document` is defined before trying to access `cookie`.

## Steps to Reproduce
1. Setup nextjs boilerplate app
2. Turn on SSR
3. Install Amplitude
5. Add page tracking
6. Run `next build`

## Environment
- JS SDK Version:
  - `@amplitude/analytics-client-common@2.0.0`
  - `@amplitude/analytics-browser@2.0.0`
- Installation Method: Yarn
- Browser and Version: Chrome 114.0.5735.106 (Official Build) (arm64)

**Comment by kevinpagtakhan** [comment]
Hi @DavidPhilip, thank you for choosing Amplitude! I couldn't reproduce on build time, but I can on runtime. I tried applying your fix and it seems to solve the issue. I'll get it merged and published shortly.

**Comment by gabrielreisn** [comment]
@kevinpagtakhan, any expectation about when it should be released on the next version?

**Comment by kevinpagtakhan** [comment]
Hi, this issue should be fixed starting v2.0.1. Thank you!
